File: spec/functional_test/fixtures/java/cli_jcommander/src/Tool.java
import com.beust.jcommander.JCommander;
import com.beust.jcommander.Parameter;
import com.beust.jcommander.Parameters;

import java.util.ArrayList;
import java.util.List;

public class Tool {
    @Parameter(names = {"-v", "--verbose"}, description = "Verbose mode")
    private boolean verbose;

    @Parameter(description = "files")
    private List<String> files = new ArrayList<>();

    public static void main(String[] args) {
        Tool tool = new Tool();
        CommandServe serve = new CommandServe();
        JCommander jc = JCommander.newBuilder()
                .addObject(tool)
                .addCommand("serve", serve)
                .build();
        jc.parse(args);
        String token = System.getenv("JC_TOKEN");
        System.out.println(token);
    }
}

@Parameters(commandNames = "serve", commandDescription = "Start the server")
class CommandServe {
    @Parameter(names = "--port", description = "Port to listen on")
    private int port = 8080;
}

File: spec/functional_test/fixtures/kotlin/cli_picocli/tool.kt
import picocli.CommandLine
import picocli.CommandLine.Command
import picocli.CommandLine.Option
import picocli.CommandLine.Parameters
import java.util.concurrent.Callable
import kotlin.system.exitProcess

@Command(name = "tool", mixinStandardHelpOptions = true, subcommands = [Serve::class])
class Tool : Callable<Int> {
    @Option(names = ["-v", "--verbose"], description = ["enable verbose logging"])
    var verbose: Boolean = false

    @Option(names = ["--config"], description = ["config path"])
    var config: String? = null

    override fun call(): Int {
        val token = System.getenv("API_TOKEN")
        return 0
    }
}

@Command(name = "serve")
class Serve : Callable<Int> {
    @Option(names = ["-p", "--port"], description = ["port to listen on"])
    var port: Int = 8080

    @Parameters(index = "0", description = ["config file"])
    lateinit var config: String

    override fun call(): Int {
        return 0
    }
}

fun main(args: Array<String>) {
    val exitCode = CommandLine(Tool()).execute(*args)
    exitProcess(exitCode)
}
